docs/development/linux-bridge-primary-nic-scenario.md

```diff
@@ -0,0 +1,787 @@
+  literal: PATH(4)
+- pos: 43
+  type: IDENTITY(1)
+  literal: running
+- pos: 50
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 51
+  type: NUMBER(2)
+  literal: "0"
+- pos: 52
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 53
+  type: IDENTITY(1)
+  literal: next-hop-interface
+- pos: 70
+  type: EOF(0)
+  literal: ""
+
+```
+
+**AST**
+```yaml
+pos: 0
+equal:
+  - pos: 0
+    path:
+      - pos: 0
+        id: interfaces
+      - pos: 11
+        id: name
+  - pos: 17
+    path:
+      - pos: 17
+        id: capture
+      - pos: 25
+        id: default-gw
+      - pos: 36
+        id: routes
+      - pos: 43
+        id: running
+      - pos: 51
+        idx: 0
+      - pos: 53
+        id: next-hop-interface
+
+```
+#### bridge-routes: capture.base-iface-routes | routes.running.next-hop-interface="br1"
+
+**Tokens**
+```yaml
+- pos: 0
+  type: IDENTITY(1)
+  literal: capture
+- pos: 7
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 8
+  type: IDENTITY(1)
+  literal: base-iface-routes
+- pos: 26
+  type: PIPE(5)
+  literal: PIPE(5)
+- pos: 28
+  type: IDENTITY(1)
+  literal: routes
+- pos: 34
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 35
+  type: IDENTITY(1)
+  literal: running
+- pos: 42
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 43
+  type: IDENTITY(1)
+  literal: next-hop-interface
+- pos: 61
+  type: ASSIGN(9)
+  literal: ASSIGN(9)
+- pos: 62
+  type: STRING(3)
+  literal: br1
+- pos: 66
+  type: EOF(0)
+  literal: ""
+- pos: 66
+  type: EOF(0)
+  literal: ""
+
+```
+
+**AST**
+```yaml
+pos: 0
+path:
+  - pos: 0
+    id: capture
+  - pos: 8
+    id: base-iface-routes
+pipe:
+    pos: 28
+    assign:
+      - pos: 28
+        path:
+          - pos: 28
+            id: routes
+          - pos: 35
+            id: running
+          - pos: 43
+            id: next-hop-interface
+      - pos: 62
+        string: br1
+
+```
+#### delete-base-iface-routes: capture.base-iface-route | routes.running.state="absent"
+
+**Tokens**
+```yaml
+- pos: 0
+  type: IDENTITY(1)
+  literal: capture
+- pos: 7
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 8
+  type: IDENTITY(1)
+  literal: base-iface-route
+- pos: 25
+  type: PIPE(5)
+  literal: PIPE(5)
+- pos: 27
+  type: IDENTITY(1)
+  literal: routes
+- pos: 33
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 34
+  type: IDENTITY(1)
+  literal: running
+- pos: 41
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 42
+  type: IDENTITY(1)
+  literal: state
+- pos: 47
+  type: ASSIGN(9)
+  literal: ASSIGN(9)
+- pos: 48
+  type: STRING(3)
+  literal: absent
+- pos: 55
+  type: EOF(0)
+  literal: ""
+- pos: 55
+  type: EOF(0)
+  literal: ""
+
+```
+
+**AST**
+```yaml
+pos: 0
+path:
+  - pos: 0
+    id: capture
+  - pos: 8
+    id: base-iface-route
+pipe:
+    pos: 27
+    assign:
+      - pos: 27
+        path:
+          - pos: 27
+            id: routes
+          - pos: 34
+            id: running
+          - pos: 42
+            id: state
+      - pos: 48
+        string: absent
+
+```
+
+***Captured state***
```

Comment:
I"m missing the part that translates the tree its result (the captured state). Something similar to [this](https://mk0tuzolorusfnc7thxk.kinstacdn.com/wp-content/uploads/2017/09/example_expression_parse_tree.png) example.